src/driver.rs

```diff
@@ -410,6 +410,90 @@ impl DriverManager {
         Ok(Driver { c_driver })
     }
 
+    /// Get the Driver based on the file extension from filename
+    ///
+    /// Searches for the available extensions in the registered
+    /// drivers and returns the matches. The determined driver is
+    /// checked for writing capabilities as
+    /// [`Dataset::open`](Dataset::open) can already open datasets
+    /// with just path.
+    ///
+    /// See also: [`get_driver_by_name`](Self::get_driver_by_name)
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// let compatible_drivers =
+    ///     DriverManager::get_drivers_for_filename("test.gpkg", true)
+    ///         .iter()
+    ///         .map(|d| d.short_name())
+    ///         .collect::<Vec<String>>();
+    /// println!("{:?}", compatible_drivers);
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// ```text
+    /// ["GPKG"]
+    /// ```
+    pub fn get_drivers_for_filename(filename: &str, is_vector: bool) -> Vec<Driver> {
+        let ext = {
+            let filename = filename.to_ascii_lowercase();
+            let e = match filename.rsplit_once('.') {
+                Some(("", _)) => "", // hidden file no ext
+                Some((f, "zip")) => {
+                    // zip files could be zipped shp or gpkg
+                    if f.ends_with(".shp") {
+                        "shp.zip"
+                    } else if f.ends_with(".gpkg") {
+                        "gpkg.zip"
+                    } else {
+                        "zip"
+                    }
+                }
+                Some((_, e)) => e, // normal file with ext
+                None => "",
+            };
+            e.to_string()
+        };
+
+        let mut drivers: Vec<Driver> = Vec::new();
+        for i in 0..DriverManager::count() {
+            let d = DriverManager::get_driver(i).expect("Index for this loop should be valid");
+            let can_create = d.metadata_item("DCAP_CREATE", "").is_some()
+                || d.metadata_item("DCAP_CREATECOPY", "").is_some();
+            let check_vector = is_vector && d.metadata_item("DCAP_VECTOR", "").is_some();
+            let check_raster = !is_vector && d.metadata_item("DCAP_RASTER", "").is_some();
+            let check_vector_translate =
+                is_vector && d.metadata_item("DCAP_VECTOR_TRANSLATE_FROM", "").is_some();
```

Comment:
See comment above... i.e. should we consider selection and filtering two different operations?